### DISCUSSION TITLE
Error when trying to upgrade PocketBase from an older version

### DISCUSSION THREAD

**Discussion by matanshamshila** [pocketbase/pocketbase#4340] [Q&A] [answered]
Hi,
I have an app in production that I built using an older version of PocketBase (exe v0.11.2).
I want to upgrade to a newer version so I can take advantage of some new features.
I am working within a closed network, so I have to transfer the files to update, I can't use the update command in the cmd.
I've tried to upgrade to a newer version of PocketBase (exe v0.20.6) by replacing the exe files, but I get this error:
"2024/02/12 08:56:35 Failed to apply migration 1677760279_uppsercase_method. go: database disk image is malformed (11)"

Would appreciate any suggestions on how I might solve this

**Comment by ganigeorgiev** [comment, marked as answer]
The `1677760279_uppsercase_method` migration is for the logs database.
To resolve the "database disk image is malformed" error you can delete the `pb_data/logs.db` file (_together with `pb_data/logs.db-wal` and `pb_data/logs.db-shm` if they exist_). The `logs.db` file will be created automatically with the next start of the application.
Why this error has happened I don't know as there are too many factors in play here - you have incorrectly transferred the files (eg. copied the db file while the application was running and missed its `-wal` side-car file or it was copied at different stages while having ongoing write), it could be a file permission error, 3rd partly application (eg. Litestream), etc.

With that said, big jumps between major release is not guarantee to work as there are some breaking changes and it is recommended to spend some time reading the changelogs (_changes marked with "⚠️" are breaking_):

- for v0.8-v15 - https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG_8_15.md
- for v15+ - https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md

You can try migrating on several steps, for example try with shorter release gap and run the binaries one by one:
1. v0.13.4 - https://github.com/pocketbase/pocketbase/releases/tag/v0.13.4
2. v0.17.7 - https://github.com/pocketbase/pocketbase/releases/tag/v0.17.7
3. v0.21.3 - https://github.com/pocketbase/pocketbase/releases/tag/v0.21.3

If you stumble on another migration script error, please report it and I'll try to identify the version that you'll need to update prior it.

**Reply by matanshamshila** [reply]
Thanks a lot
I removed the pb_data/logs.db file (together with pb_data/logs.db-wal and pb_data/logs.db-shm) and now it works

Do you recommend to try migrating on several steps anyway, or I can use the latest version as long as everything appears to be working well?

**Reply by ganigeorgiev** [reply]
If you are able to run the latest binary then you don't have to upgrade on multiple steps.

But note that while the system migrations attempts to minimize the breaking changes related to the stored data, there could be still some changes that may require adjusting your custom code (_most of the breaking changes are related to the exposed Go interfaces so if you are not using PocketBase as framework most likely you'll not have to do anything further_).
